smithy-model/src/main/resources/software/amazon/smithy/model/loader/prelude.smithy

```diff
@@ -191,7 +191,44 @@ map externalDocumentation {
 }
 
 /// Defines the ordered list of supported authentication schemes.
-@trait(selector: ":is(service, operation)")
+@trait(
+    selector: ":is(service, operation)"
+    breakingChanges: [
+        {
+            change: "add"
+            severity: "DANGER"
+            message: """
+                Adding the `@auth` trait will explicitly define the order of the \
+                authentication scheme traits for the target shape. \
+                If the target shape is a service, all contained operations will \
+                inherit the same `@auth` trait unless a contained operation has \
+                its own `@auth` trait applied."""
+        }
+        {
+            change: "remove"
+            severity: "DANGER"
+            message: """
+                Removing the `@auth` trait will default to using the alphabetical \
+                order of the absolute shape IDs of each authentication scheme trait \
+                for the target shape. \
+                If the target shape is a service, all contained operations will \
+                inherit the same alphabetical order unless a contained operation \
+                has its own `@auth` trait applied."""
+        }
+        {
+            change: "update"
+            severity: "DANGER"
+            message: """
+                Since the `@auth` trait is priority ordered, updating the order of \
+                the authentication schemes may break authentication for clients \
+                and servers, especially if the first authentication scheme is \
```

Comment:
what is special about the first auth scheme?

Reply:
Because `@auth` is priority-ordered, breaking changes could occur in edge cases.

It probably does not need to be explicitly called out in the generic case.

Reply:
Note: the `@auth` trait diff validation changes will be split into a different PR.

Reply:
Changed in #2252